src/meltano/core/error.py

```diff
@@ -83,102 +53,8 @@ async def stderr(self) -> Optional[str]:
 
 
 class PluginInstallError(Exception):
-    """Happens when a plugin fails to install."""
+    """Exception for when a plugin fails to install."""
 
 
 class PluginInstallWarning(Exception):
-    """Happens when a plugin optional optional step fails to install."""
-
-
-class PluginNotInstallable(Exception):
-    pass
-
-
-def aggregate(error_cls):
-    class Aggregate(error_cls):
-        """Aggregate multiple sub-exceptions."""
-
-        def __init__(self, exceptions: []):
-            self.exceptions = exceptions
-
-        def __str__(self):
-            return "\n".join(str(e) for e in self.exceptions)
-
-    if error_cls != Exception:
-        error_cls.Aggregate = Aggregate
-
-    return error_cls
```

Comment:
First time seeing taking a look at this code. Interesting that Meltano devs were trying what's roughly [PEP 654](https://peps.python.org/pep-0654/).

Reply:
It's a good idea, but difficult to do without library/language support. Even with library/language support, it's difficult to go from the current exception model to this new one.